bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/discovery/AbstractMQTTDiscovery.java

```diff
@@ -48,11 +48,34 @@
 
     private @Nullable ScheduledFuture<?> scheduledStop;
 
+    private boolean isSubscribed;
+
     public AbstractMQTTDiscovery(@Nullable Set<ThingTypeUID> supportedThingTypes, int timeout,
             boolean backgroundDiscoveryEnabledByDefault, String baseTopic) {
         super(supportedThingTypes, 0, backgroundDiscoveryEnabledByDefault);
         this.subscribeTopic = baseTopic;
         this.timeout = timeout;
+        isSubscribed = false;
+    }
+
+    /**
+     * Only subscribe if we were not already subscribed
+     */
+    private void subscribe() {
+        if (!isSubscribed) {
+            getDiscoveryService().subscribe(this, subscribeTopic);
+            isSubscribed = true;
+        }
+    }
+
+    /**
+     * Only unsubscribe if we were already subscribed
+     */
+    private void unSubscribe() {
+        if (isSubscribed) {
```

Comment:
if (isSubscribed.getAndSet(false)) {
    getDiscoveryService().unsubscribe(this);
}

Reply:
Ok.